counter/src/main/java/org/infinispan/counter/impl/strong/AbstractStrongCounter.java

```diff
@@ -124,7 +124,12 @@ public final String getName() {
 
    @Override
    public CompletableFuture<Boolean> compareAndSet(long expect, long update) {
-      return readWriteMap.eval(key, new CompareAndSetFunction<>(expect, update))
+      return compareAndSwap(expect, update).thenApply(value -> value == expect);
```

Comment:
Have you considered moving this to interface as a default method?

Reply:
totally forgot about default methods :+1: